src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/compute/v2/OpenStackComputePluginTest.java

```diff
@@ -1,858 +1,688 @@
+    // test case: when a request is unsuccessful, it should thrown UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testDeleteInstanceUnsuccessful() throws FogbowException, HttpResponseException {
         // set up
-        List<String> networksId = null;
-        ComputeOrder computeOrder = new ComputeOrder(null, null, null, "default", null, bestCpu, bestMemory, bestDisk, imageId, null,
-                publicKey, networksId);
-        responseNetworkIds.remove(this.privateNetworkId);
-
-        JSONObject computeJson = generateJsonRequest(imageId, bestFlavorId, userData, idKeyName, responseNetworkIds,
-                idInstanceName);
-
-        mockGetFlavorsRequest(bestFlavorId, bestCpu, bestMemory, bestDisk);
-        Mockito.when(this.clientMock.doPostRequest(this.argString.capture(), this.argBodyString.capture(), this.argCloudToken.capture()
-        )).thenReturn("");
-        Mockito.when(this.launchCommandGeneratorMock.createLaunchCommand(computeOrder)).thenReturn(userData);
-        Mockito.doReturn(idKeyName).doReturn(idInstanceName).when(this.computePlugin).getRandomUUID();
-        Mockito.doReturn(expectedInstanceIdJson).when(this.clientMock).doPostRequest(argString.capture(),
-                argBodyString.capture(), argCloudToken.capture());
+        ComputeOrder computeOrder = testUtils.createLocalComputeOrder();
+        Mockito.doThrow(HttpResponseException.class)
+                .when(this.clientMock).doDeleteRequest(Mockito.any(), Mockito.any());
 
         // exercise
-        String instanceId = this.computePlugin.requestInstance(computeOrder, this.openStackV3User);
-
-        // verify
-        Assert.assertEquals(this.osKeyPairEndpoint, this.argString.getAllValues().get(0));
-        Assert.assertEquals(this.openStackV3User, this.argCloudToken.getAllValues().get(0));
-        JSONAssert.assertEquals(rootKeypairJson.toString(), this.argBodyString.getAllValues().get(0).toString(), false);
+        this.computePlugin.deleteInstance(computeOrder, cloudUser);
 
-        Assert.assertEquals(computeEndpoint, this.argString.getAllValues().get(1));
-        Assert.assertEquals(this.openStackV3User, this.argCloudToken.getAllValues().get(1));
-        JSONAssert.assertEquals(computeJson.toString(), this.argBodyString.getAllValues().get(1).toString(), false);
-
-        Assert.assertEquals(expectedInstanceId, instanceId);
+        Assert.fail();
     }
 
-    // test case: Request Instance should throw InvalidParameter if the Get Request on updateFlavors returns Bad Request
-    // while getting flavors id
-    @Test(expected = InvalidParameterException.class)
-    public void testRequestInstanceWhenFlavorsIdRequestException() throws HttpResponseException, FogbowException {
+    // test case: Test if after a successful request, it can return the
+    // appropriate computeInstance
+    @Test
+    public void testGetInstanceSuccessful() throws FogbowException, HttpResponseException {
```

Comment:
getInstance can also be refactored to match our pattern. For example, extract some lines to a doGetInstance